core/modules/tests.rs

```diff
@@ -2175,3 +2175,208 @@ async fn evaluate_already_evaluated_module_sync() {
     .execute_script("check2", "if (globalThis.syncExecutionCount !== 1) throw new Error('Expected still 1 execution')")
     .unwrap();
 }
+
+#[tokio::test]
+async fn test_native_source_map_with_trailing_content() {
+  // This test verifies that source maps work even with trailing newlines and comments
+  // after the sourceMappingURL directive (https://github.com/denoland/deno/issues/21988)
+
+  let source_map_base64 = "eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImZpbGU6Ly8vdGVzdC50cyJdLCJzb3VyY2VzQ29udGVudCI6WyJmdW5jdGlvbiBncmVldChuYW1lOiBzdHJpbmcpIHtcbiAgY29uc29sZS5sb2coXCJIZWxsbywgXCIgKyBuYW1lKTtcbn1cblxuZ3JlZXQoXCJXb3JsZFwiKTtcbiJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtBQUNBO0FBQ0E7QUFDQTtBQUNBIn0=";
+
+  let code = format!(
+    r#"export function greet(name) {{
+  console.log("Hello, " + name);
+}}
+
+greet("World");
```

Comment:
I think this code needs to throw an error to test it fully - otherwise we're not checking if the source map is actually getting applied, right?